### DISCUSSION TITLE
plz kelp - python - create AI<title>

### DISCUSSION THREAD

**Discussion by Frankgtv6** [community/community#118471] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

hi

why its not working ?

from keras import *

model = Sequential()

model.add(layers.Dense(units=3, input_shape=[1]))

model.add(layers.Dense(units=64,))

model.add(layers.Dense(units=1,))

entree = [2, 4, 6]
sortie = [1, 2, 3]

model.compile(loss='mean_squared_error', optimizer='adam')

model.fit(x=entree, y=sortie, epochs=100)

while True:
    x = int(input('Nombre :'))
    print('Prediction : ' + str(model.predict([x])))

![AII](https://github.com/community/community/assets/162024865/236f854a-14e6-43e6-b7dc-e98c6c734eb6)

**Comment by jfullstackdev** [comment]
Copilot tried to answer this

---

The code you've provided seems to be a simple neural network model built with Keras. It's designed to predict output based on the input it's trained on. However, without a specific error message or description of the problem, it's hard to provide a precise solution.

That being said, there are a few potential issues that could arise with this code:

1. **Indentation**: Python is sensitive to indentation. The last two lines of your code should be indented as they are part of the while loop:

```python
while True:
    x = int(input('Nombre :'))
    print('Prediction : ' + str(model.predict([x])))
```

2. **Data Shape**: The `model.predict()` function expects a 2D array-like input. If `x` is a single integer, you need to reshape it into a 2D array before passing it to `model.predict()`. You can do this with numpy:

```python
import numpy as np

while True:
    x = int(input('Nombre :'))
    x = np.array([x]).reshape(-1, 1)
    print('Prediction : ' + str(model.predict(x)))
```

3. **Model Training**: You're training the model on a very small dataset (only three samples), and for a relatively small number of epochs. This might not be enough for the model to learn accurately, especially if the relationship between the input and output is complex.

If you're still having trouble, please provide more details about the problem, such as any error messages you're seeing, and I'd be happy to help further.
